### PR TITLE
Implementation for SBOL2<-->SBOL3 converter

### PR DESCRIPTION
**Pull request by Yehuda-Binik**
In the new converter, I completed the `visit_implementation` function for both the `SBOL3To2ConversionVisitor` and `SBOL2To3ConversionVisitor` classes. I also added two test cases to test my code using the SBOL3 [implementation.nt](https://github.com/SynBioDex/SBOLTestSuite/blob/master/SBOL3/entity/implementation/implementation.nt) file from the [SBOLTestSuite](https://github.com/SynBioDex/SBOLTestSuite/tree/master).

### CODE REVIEW

**Comment by jakebeal** [comment]
@Yehuda-Binik : there are some files that shouldn't have been checked in that were checked in. Can you please remove the `.idea` and `sbol_utilities.egg-info` directories?  In general, `.idea` and `*.egg-info` should be in your `.gitignore` files.

**Comment by Yehuda-Binik** [comment]
@jakebeal I removed the `.idea` and `sbol_utilities.egg-info` directories. I also updated my `.gitignore` Is it necessary to delete my `.gitignore` file from the pull request as well? Let me know if there is anything left I need to change.

**Comment by jakebeal** [comment]
Thank you! Adding to .gitignore is fine and helpful. I will review the PR in detail now.

File: sbol_utilities/sbol3_sbol2_conversion.py [hunk 1]
```diff
@@ -462,9 +467,15 @@ def visit_generic_location(self, a: sbol2.GenericLocation):
...
+    def visit_implementation(self, imp2: sbol2.Implementation):
         # Priority: 1
-        raise NotImplementedError('Conversion of Implementation from SBOL2 to SBOL3 not yet implemented')
+        # Make the Implementation object and add it to the document
+        imp3 = sbol3.Implementation(imp2.identity, namespace=self._sbol3_namespace(imp2), built=imp2.built)
+        self.doc3.add(imp3)
+        # Map over all other TopLevel properties and extensions not covered by the constructor
+        self._convert_toplevel(imp2, imp3)
+
+        # raise NotImplementedError('Conversion of Implementation from SBOL2 to SBOL3 not yet implemented')
```
Comment (jakebeal) [inline sbol_utilities/sbol3_sbol2_conversion.py]: Delete this comment as unneeded.

File: test/sbol3.nt
Comment (jakebeal) [inline test/sbol3.nt:1]: This file is out of place: it's in the test directory, not the test_files subdirectory. It looks like an accidental check-in?

File: test/test_files/sbol3_genbank_conversion/ignoring_sbol_properties.nt.test
Comment (jakebeal) [inline test/test_files/sbol3_genbank_conversion/ignoring_sbol_properties.nt.test:1]: This file also appears to be an accidental check-in.

File: test/test_sbol2_sbol3_direct.py [hunk 2]
```diff
@@ -11,15 +11,15 @@
 
 TEST_FILES = Path(__file__).parent / 'test_files'
 
-
```
Comment (jakebeal) [inline test/test_sbol2_sbol3_direct.py]: Please avoid making un-needed changes in spacing: it increases the likelihood of conflicts.
There are several locations, here and below.

File: test/test_sbol2_sbol3_direct.py [hunk 3]
```diff
@@ -35,8 +35,8 @@ def test_3to2_conversion(self):
             self.assertFalse(file_diff(str(tmp3), str(TEST_FILES / 'BBa_J23101_patched.nt')))
 
     def test_2to3_conversion(self):
-        """Test ability to convert a simple part from SBOL3 to SBOL2"""
-        # Load an SBOL3 document and check its contents
+        """Test ability to convert a simple part from SBOL2 to SBOL3"""
```
Comment (jakebeal) [inline test/test_sbol2_sbol3_direct.py:38]: Good catch!

**Review by jakebeal — CHANGES_REQUESTED** [review]
There's some minor cleanup needed - the substance of the changes is all good, I just need you to remove the accidental check-ins and unnecessary space changes.

**Comment by Yehuda-Binik** [comment]
I removed the two accidentally checked-in files deleted 3 unnecessary comments (the unneeded `raise NotImplemented` comment as well as two comments I had written myself) and removed superfluous formatting changes.

**Review by jakebeal — APPROVED** [review]
Code looks good, and the two errors are, I believe, only from the secret-requiring tests that are not affected by this PR

**Pull request merged**